[8.9] Legacy Elasticsearch exporter rendered when OpenSearch is the secondary storage with Optimize

## Description

The helper `orchestration.hasLegacyElasticsearchExporter` in `templates/orchestration/_helpers.tpl` renders the legacy Elasticsearch exporter whenever both `orchestration.exporters.rdbms.enabled` and `optimize.enabled` are true, without checking whether Elasticsearch is actually enabled. The faulty branch is:

```gotemplate
(and .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
```

With no `global.elasticsearch.enabled` guard, any RDBMS + Optimize values file gets the ES exporter injected into Zeebe broker config — even when Elasticsearch is disabled and OpenSearch is the only configured backend. The broker then tries to connect to `<release>-elasticsearch:9200` and fails.

## Expected vs Actual Behavior

**Expected:** with `orchestration.exporters.rdbms.enabled=true`, `optimize.enabled=true`, and `global.elasticsearch.enabled=false`, the legacy ES exporter is **not** rendered into `zeebe.broker.exporters`.

**Actual:** the legacy ES exporter renders into the orchestration ConfigMap:

```yaml
exporters:
  elasticsearch:
    className: "io.camunda.zeebe.exporter.ElasticsearchExporter"
    args:
      url: "http://<release>-elasticsearch:9200"
```

Broker log: `Caused by: java.net.UnknownHostException: <release>-elasticsearch: Name or service not known`

## Steps to Reproduce

```yaml
global:
  elasticsearch:
    enabled: false
  opensearch:
    enabled: true
orchestration:
  exporters:
    rdbms:
      enabled: true
    zeebe:
      enabled: true
optimize:
  enabled: true
  database:
    elasticsearch:
      enabled: false
    opensearch:
      enabled: true
```

`helm template integration charts/camunda-platform-8.9 -f values.yaml | grep -c ElasticsearchExporter` → `1` (should be `0`).

## Acceptance Criteria

- [x] Legacy ES exporter suppressed when `rdbms+optimize` enabled but `global.elasticsearch.enabled=false`.
- [x] Existing ES + Optimize scenarios still render ES exporter (no regression).
- [x] Unit test covers `rdbms + optimize + zeebe + OpenSearch, ES disabled`.
- [x] Fix applied to both `camunda-platform-8.9` and `camunda-platform-8.10`.

## Additional Context

Fix: gate branch A on `global.elasticsearch.enabled` in `_helpers.tpl`:

```gotemplate
(and .Values.global.elasticsearch.enabled .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
```

Regression test added: `TestLegacyESExporterAbsentForSupport32901CustomerConfig` in `test/unit/orchestration/configmap_unified_test.go` (both versions).

## References

- Jira: [SUPPORT-32901](https://jira.camunda.com/browse/SUPPORT-32901)
- Incident: INC-5695
- Faulty line (8.9): https://github.com/camunda/camunda-platform-helm/blob/77ff360ba0e8dd95766acdfd881c73bfa22bdc45/charts/camunda-platform-8.9/templates/orchestration/_helpers.tpl#L293
- Same helper in 8.10: `charts/camunda-platform-8.10/templates/orchestration/_helpers.tpl#L290-L304`

## Environment

- **Platform:** n/a (chart rendering bug)
- **Helm CLI version:** n/a
- **Chart version:** `camunda-platform-8.9`, `camunda-platform-8.10`
- **Kubernetes version:** n/a

--- a/charts/camunda-platform-8.10/templates/orchestration/_helpers.tpl
+++ b/charts/camunda-platform-8.10/templates/orchestration/_helpers.tpl
@@ -289,8 +289,8 @@ and
 
 {{- define "orchestration.hasLegacyElasticsearchExporter" -}}
 {{- and
-      (or 
-        (and .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
+      (or
+        (and .Values.global.elasticsearch.enabled .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
         (or
           (and .Values.global.elasticsearch.enabled .Values.orchestration.exporters.zeebe.enabled)
           (and (or .Values.global.elasticsearch.enabled .Values.optimize.database.elasticsearch.enabled) .Values.optimize.enabled)

--- a/charts/camunda-platform-8.10/test/unit/orchestration/configmap_unified_test.go
+++ b/charts/camunda-platform-8.10/test/unit/orchestration/configmap_unified_test.go
@@ -494,6 +494,110 @@ func (s *ConfigmapTemplateTest) TestDifferentValuesInputsUnifiedRDBMS() {
 	testhelpers.RunTestCases(s.T(), s.chartPath, s.release, s.namespace, s.templates, testCases)
 }
 
+func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
+	testCases := []testhelpers.TestCase{
+		{
+			Name: "TestLegacyESExporterAbsentWhenRdbmsAndOptimizeButNoElasticsearch",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.opensearch.enabled":                          "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize without elasticsearch must not render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterPresentWhenRdbmsAndOptimizeAndGlobalElasticsearch",
+			Values: map[string]string{
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.Contains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize with global.elasticsearch.enabled must render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterPresentWhenRdbmsAndOptimizeDatabaseElasticsearchOnly",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.elasticsearch.enabled":                       "true",
+				"optimize.database.elasticsearch.external":                      "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.Contains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize with optimize.database.elasticsearch.enabled must render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterAbsentForSupport32901CustomerConfig",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.exporters.zeebe.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.opensearch.enabled":                          "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"SUPPORT-32901: rdbms+optimize+zeebe with OpenSearch must not render legacy ES exporter")
+				require.Contains(t, output, "io.camunda.zeebe.exporter.opensearch.OpensearchExporter",
+					"SUPPORT-32901: OS exporter must still render to feed Optimize")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterAbsentWhenOnlyRdbmsNoOptimize",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms without optimize must not render legacy ES exporter")
+			},
+		},
+	}
+
+	testhelpers.RunTestCasesE(s.T(), s.chartPath, s.release, s.namespace, s.templates, testCases)
+}
+
 func (s *ConfigmapTemplateTest) TestMultiRegionInitialContactPoints() {
 	testCases := []testhelpers.TestCase{
 		{

--- a/charts/camunda-platform-8.10/test/unit/orchestration/configmap_unified_test.go
+++ b/charts/camunda-platform-8.10/test/unit/orchestration/configmap_unified_test.go
@@ -500,7 +500,6 @@ func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
 			Name: "TestLegacyESExporterAbsentWhenRdbmsAndOptimizeButNoElasticsearch",
 			Values: map[string]string{
 				"global.elasticsearch.enabled":                                  "false",
-				"elasticsearch.enabled":                                         "false",
 				"global.opensearch.enabled":                                     "true",
 				"global.opensearch.url.host":                                    "opensearch.example.com",
 				"orchestration.exporters.rdbms.enabled":                         "true",
@@ -535,7 +534,6 @@ func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
 			Name: "TestLegacyESExporterPresentWhenRdbmsAndOptimizeDatabaseElasticsearchOnly",
 			Values: map[string]string{
 				"global.elasticsearch.enabled":                                  "false",
-				"elasticsearch.enabled":                                         "false",
 				"global.opensearch.enabled":                                     "true",
 				"global.opensearch.url.host":                                    "opensearch.example.com",
 				"orchestration.exporters.rdbms.enabled":                         "true",
@@ -556,7 +554,6 @@ func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
 			Name: "TestLegacyESExporterAbsentForSupport32901CustomerConfig",
 			Values: map[string]string{
 				"global.elasticsearch.enabled":                                  "false",
-				"elasticsearch.enabled":                                         "false",
 				"global.opensearch.enabled":                                     "true",
 				"global.opensearch.url.host":                                    "opensearch.example.com",
 				"orchestration.exporters.rdbms.enabled":                         "true",
@@ -579,7 +576,6 @@ func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
 			Name: "TestLegacyESExporterAbsentWhenOnlyRdbmsNoOptimize",
 			Values: map[string]string{
 				"global.elasticsearch.enabled":                                  "false",
-				"elasticsearch.enabled":                                         "false",
 				"global.opensearch.enabled":                                     "true",
 				"global.opensearch.url.host":                                    "opensearch.example.com",
 				"orchestration.exporters.rdbms.enabled":                         "true",

--- a/charts/camunda-platform-8.9/templates/orchestration/_helpers.tpl
+++ b/charts/camunda-platform-8.9/templates/orchestration/_helpers.tpl
@@ -289,8 +289,8 @@ and
 
 {{- define "orchestration.hasLegacyElasticsearchExporter" -}}
 {{- and
-      (or 
-        (and .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
+      (or
+        (and .Values.global.elasticsearch.enabled .Values.orchestration.exporters.rdbms.enabled .Values.optimize.enabled)
         (or
           (and .Values.global.elasticsearch.enabled .Values.orchestration.exporters.zeebe.enabled)
           (and (or .Values.global.elasticsearch.enabled .Values.optimize.database.elasticsearch.enabled) .Values.optimize.enabled)

--- a/charts/camunda-platform-8.9/test/unit/orchestration/configmap_unified_test.go
+++ b/charts/camunda-platform-8.9/test/unit/orchestration/configmap_unified_test.go
@@ -513,6 +513,110 @@ func (s *ConfigmapTemplateTest) TestDifferentValuesInputsUnifiedRDBMS() {
 	testhelpers.RunTestCases(s.T(), s.chartPath, s.release, s.namespace, s.templates, testCases)
 }
 
+func (s *ConfigmapTemplateTest) TestHasLegacyElasticsearchExporter() {
+	testCases := []testhelpers.TestCase{
+		{
+			Name: "TestLegacyESExporterAbsentWhenRdbmsAndOptimizeButNoElasticsearch",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.opensearch.enabled":                          "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize without elasticsearch must not render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterPresentWhenRdbmsAndOptimizeAndGlobalElasticsearch",
+			Values: map[string]string{
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.Contains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize with global.elasticsearch.enabled must render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterPresentWhenRdbmsAndOptimizeDatabaseElasticsearchOnly",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.elasticsearch.enabled":                       "true",
+				"optimize.database.elasticsearch.external":                      "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.Contains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms+optimize with optimize.database.elasticsearch.enabled must render legacy ES exporter")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterAbsentForSupport32901CustomerConfig",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.exporters.zeebe.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+				"optimize.enabled":                                              "true",
+				"optimize.database.opensearch.enabled":                          "true",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"SUPPORT-32901: rdbms+optimize+zeebe with OpenSearch must not render legacy ES exporter")
+				require.Contains(t, output, "io.camunda.zeebe.exporter.opensearch.OpensearchExporter",
+					"SUPPORT-32901: OS exporter must still render to feed Optimize")
+			},
+		},
+		{
+			Name: "TestLegacyESExporterAbsentWhenOnlyRdbmsNoOptimize",
+			Values: map[string]string{
+				"global.elasticsearch.enabled":                                  "false",
+				"elasticsearch.enabled":                                         "false",
+				"global.opensearch.enabled":                                     "true",
+				"global.opensearch.url.host":                                    "opensearch.example.com",
+				"orchestration.exporters.rdbms.enabled":                         "true",
+				"orchestration.data.secondaryStorage.rdbms.url":                 "jdbc:postgresql://localhost:5432/camunda",
+				"orchestration.data.secondaryStorage.rdbms.username":            "camunda",
+				"orchestration.data.secondaryStorage.rdbms.secret.inlineSecret": "my-password",
+			},
+			Verifier: func(t *testing.T, output string, err error) {
+				require.NoError(t, err)
+				require.NotContains(t, output, "io.camunda.zeebe.exporter.ElasticsearchExporter",
+					"rdbms without optimize must not render legacy ES exporter")
+			},
+		},
+	}
+
+	testhelpers.RunTestCasesE(s.T(), s.chartPath, s.release, s.namespace, s.templates, testCases)
+}
+
 func (s *ConfigmapTemplateTest) TestMultiRegionInitialContactPoints() {
 	testCases := []testhelpers.TestCase{
 		{

